File: Year 3/React Native/Simple SMHI Weather Warning application/my-second-project/models/Geometry.ts
export class Geometry {
    private type: string | null;
    private pointCoordinates: any[];
    private lineStringCoordinates: any[];
    private polygonCoordinates: any[];
    private multiPolygonCoordinates: any[];

    constructor(type: string | null = null, 
                pointCoordinates: any[] = [], 
                lineStringCoordinates: any[] = [], 
                polygonCoordinates: any[] = [], 
                multiPolygonCoordinates: any[] = []) {
        this.type = type;
        this.pointCoordinates = pointCoordinates;
        this.lineStringCoordinates = lineStringCoordinates;
        this.polygonCoordinates = polygonCoordinates;
        this.multiPolygonCoordinates = multiPolygonCoordinates;
    }

    getType(): string | null {
        return this.type;
    }

    setType(type: string | null): void {
        this.type = type;
    }

    getPointCoordinates(): any[] {
        return this.pointCoordinates;
    }

    setPointCoordinates(pointCoordinates: any[]): void {
        this.pointCoordinates = pointCoordinates;
    }

    getLineStringCoordinates(): any[] {
        return this.lineStringCoordinates;
    }

    setLineStringCoordinates(lineStringCoordinates: any[]): void {
        this.lineStringCoordinates = lineStringCoordinates;
    }

    getPolygonCoordinates(): any[] {
        return this.polygonCoordinates;
    }

    setPolygonCoordinates(polygonCoordinates: any[]): void {
        this.polygonCoordinates = polygonCoordinates;
    }

    getMultiPolygonCoordinates(): any[] {
        return this.multiPolygonCoordinates;
    }

    setMultiPolygonCoordinates(multiPolygonCoordinates: any[]): void {
        this.multiPolygonCoordinates = multiPolygonCoordinates;
    }
}

File: Year 3/Android Java/Simple SMHI Warning App JavaVersion/GeoWarning/app/src/main/java/com/example/geowarning/model/Geometry.java
package com.example.geowarning.model;

import java.util.List;

public class Geometry {
    private GeometryTypeEnum type;
    private List<Double> pointCoordinates;
    private List<Double> lineStringCoordinates;
    private List<List<Double>> polygonCoordinates;
    private List<List<List<Double>>> multiPolygonCoordinates;

    public GeometryTypeEnum getType() {
        return type;
    }

    public void setType(GeometryTypeEnum type) {
        this.type = type;
    }

    public List<Double> getPointCoordinates() {
        return pointCoordinates;
    }

    public void setPointCoordinates(List<Double> pointCoordinates) {
        this.pointCoordinates = pointCoordinates;
    }
    public List<Double> getLineStringCoordinates() {
        return lineStringCoordinates;
    }

    public void setLineStringCoordinates(List<Double> lineStringCoordinates) {
        this.lineStringCoordinates = lineStringCoordinates;
    }

    public List<List<Double>> getPolygonCoordinates() {
        return polygonCoordinates;
    }

    public void setPolygonCoordinates(List<List<Double>> polygonCoordinates) {
        this.polygonCoordinates = polygonCoordinates;
    }

    public List<List<List<Double>>> getMultiPolygonCoordinates() {
        return multiPolygonCoordinates;
    }

    public void setMultiPolygonCoordinates(List<List<List<Double>>> multiPolygonCoordinates) {
        this.multiPolygonCoordinates = multiPolygonCoordinates;
    }
}
